electricitymap/capacity_parsers/EMBER.py

```diff
@@ -0,0 +1,176 @@
+    return df
+
+
+def format_ember_data(df: pd.DataFrame, year: int) -> pd.DataFrame:
+    df_filtered = df.loc[df["Area type"] == "Country"].copy()
+    df_filtered = df_filtered.loc[df_filtered["Year"] == year]
+    if df_filtered.empty:
+        raise ValueError(f"No data for year {year}")
+    df_filtered = df_filtered.loc[
+        (df_filtered["Category"] == "Capacity") & (df_filtered["Subcategory"] == "Fuel")
+    ]
+    # filter out Kosovo because it is not a country in pycountry
+    df_filtered = df_filtered.loc[df_filtered["Area"] != "Kosovo"]
+
+    df_filtered["country_code_iso2"] = df_filtered["Country code"].apply(
+        lambda x: pycountry.countries.get(alpha_3=x).alpha_2
+    )
+    df_filtered = df_filtered.loc[df_filtered["country_code_iso2"].isin(EMBER_ZONES)]
+
+    df_capacity = df_filtered[["country_code_iso2", "Year", "Variable", "Value"]]
+    df_capacity = df_capacity.rename(
+        columns={
+            "country_code_iso2": "zone_key",
+            "Year": "datetime",
+            "Variable": "mode",
+            "Value": "value",
+        }
+    )
+    df_capacity["datetime"] = df_capacity["datetime"].apply(lambda x: datetime(x, 1, 1))
+    df_capacity["value"] = df_capacity["value"] * 1000
+
+    df_capacity = df_capacity.apply(map_variable_to_mode, axis=1)
+
+    df_capacity = (
+        df_capacity.groupby(["zone_key", "datetime", "mode"])[["value"]]
+        .sum()
+        .reset_index()
+        .set_index(["zone_key"])
+    )
+    return df_capacity
+
+
+def get_capacity_dict_from_df(df_capacity: pd.DataFrame) -> dict:
+    all_capacity = {}
+    for zone in df_capacity.index.unique():
+        df_zone = df_capacity.loc[zone]
+        zone_capacity = {}
+        for i, data in df_zone.iterrows():
+            mode_capacity = {}
+            mode_capacity["datetime"] = data["datetime"].strftime("%Y-%m-%d")
+            mode_capacity["value"] = round(float(data["value"]), 0)
+            mode_capacity["source"] = "Ember, Yearly electricity data"
+            zone_capacity[data["mode"]] = mode_capacity
+        all_capacity[zone] = zone_capacity
+    return all_capacity
+
+
+def fetch_production_capacity_for_all_zones(target_datetime: datetime) -> dict:
+    df_capacity = get_data_from_url()
+    df_capacity = format_ember_data(df_capacity, target_datetime.year)
+    all_capacity = get_capacity_dict_from_df(df_capacity)
+    print(f"Fetched capacity data from Ember for {target_datetime.year}")
+    return all_capacity
+
+
+def fetch_production_capacity(target_datetime: datetime, zone_key: ZoneKey) -> dict:
+    all_capacity = fetch_production_capacity_for_all_zones(target_datetime)
+    if zone_key in all_capacity:
+        zone_capacity = all_capacity[zone_key]
+        print(
+            f"Fetched capacity for {zone_key} in {target_datetime.year}: \n {zone_capacity}"
+        )
+        return zone_capacity
+    else:
+        raise ValueError(f"No capacity data for {zone_key} in {target_datetime.year}")
```

Comment:
Same as with IRENA, we might need to standardise the way we handle exceptions to all the capacity parsers. Do we want to raise an Exception or we want to log it as a warning?